File: java/src/main/java/org/umundo/model/Scoreboard.java
package org.umundo.model;

import org.umundo.core.Message;

import java.util.HashMap;
import java.util.Map;

public class Scoreboard {

  private HashMap<String, Integer> scores;

  public Scoreboard(HashMap<String, Integer> scores) {
    this.scores = scores;
  }

  public HashMap<String, Integer> getScores() {
    return scores;
  }

  public Message get() {
    Message m = new Message();
    m.putMeta("type", "score");
    for(Map.Entry<String, Integer> e : scores.entrySet()) {
      m.putMeta(e.getKey(), e.getValue().toString());
    }
    return m;
  }

  public static Scoreboard fromMessage(Message m) {
    HashMap<String, Integer> scores = new HashMap<>();
    for(int i =0; i < m.getMetaKeys().size(); i++) {
      String key = m.getMetaKeys().get(i);
      if (!key.equals("type") && !key.startsWith("um.")) {
        scores.put(key, Integer.parseInt(m.getMeta(key)));
      }
    }
    return new Scoreboard(scores);
  }
}

File: pyQuiz/scoreboard.py
class Scoreboard:
    def __init__(self, client):
        self._client = client
        self._scores = {
            self._client.ui.username: 0
        }

    def dispatchWelcome(self, msg):
        username = msg.getMeta("username")

        if username not in self._scores:
            self._scores[username] = 0

    def dispatchScores(self, msg):
        scores = dict((user, score) for (user, score) in msg.getMetaKeys() if user != 'type' and not user.startswith('um.'))
        print(scores)
        self._client.ui.updateScores(scores)
